src/mono/mono/metadata/debug-mono-ppdb.c

```diff
@@ -87,6 +87,15 @@ get_pe_debug_info (MonoImage *image, guint8 *out_guid, gint32 *out_age, gint32 *
 		debug_dir.address         = read32(data + 20);
 		debug_dir.pointer         = read32(data + 24);
 
+		if (pdb_checksum_hash_type && pdb_checksum && debug_dir.type == DEBUG_DIR_PDB_CHECKSUM)
+		{
+			data  = (guint8 *) (image->raw_data + debug_dir.pointer);
+			char* alg_name = (char*)data;
+			guint8*	checksum = (guint8 *) (data + strlen(alg_name)+ 1);
```

Comment:
Should we have a bounds-check here?

Reply:
I'm not sure, this data will be used on buffer_add_string which will access until find a \0, right?

Reply:
Do you mean a case where there isn't a \0 and buffer_add_string will read it and pass the end of the debug info?
I'm not sure if it's possible but I can check if you think we should :)